# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xorg-server/hotfix-misavescreen.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- b/hw/vfb/InitOutput.c
+++ a/hw/vfb/InitOutput.c
@@ -461,6 +461,12 @@
     }
 }
 
+static Bool
+vfbSaveScreen(ScreenPtr pScreen, int on)
+{
+    return TRUE;
+}
+
 #ifdef HAVE_MMAP
 
 /* this flushes any changes to the screens out to the mmapped file */
@@ -929,6 +935,8 @@
        return FALSE;
 
     pScreen->InstallColormap = vfbInstallColormap;
+
+    pScreen->SaveScreen = vfbSaveScreen;
     pScreen->StoreColors = vfbStoreColors;
 
     miDCInitialize(pScreen, &vfbPointerCursorFuncs);
--- xorg-server-21.1.11/mi/miscrinit.c.vanilla	2024-03-10 12:50:25.594267854 +0100
+++ xorg-server-21.1.11/mi/miscrinit.c	2024-03-10 12:50:34.373268307 +0100
@@ -124,12 +124,6 @@
     return ((*pScreen->DestroyPixmap) ((PixmapPtr) pScreen->devPrivate));
 }
 
-static Bool
-miSaveScreen(ScreenPtr pScreen, int on)
-{
-    return TRUE;
-}
-
 void
 miSourceValidate(DrawablePtr pDrawable, int x, int y, int w, int h,
                  unsigned int subWindowMode)
@@ -256,7 +250,6 @@
     }
     /* else CloseScreen */
     /* QueryBestSize */
-    pScreen->SaveScreen = miSaveScreen;
     /* GetImage, GetSpans */
     pScreen->SourceValidate = miSourceValidate;
     /* CreateWindow, DestroyWindow, PositionWindow, ChangeWindowAttributes */
